### PR TITLE
added support for Component Library on Component Instances

### PR DESCRIPTION
**Pull request by mizrael**
added the `ComponentLibraryUniqueName` on Component Instances

image

### CODE REVIEW

File: src/Persistence.Tests/_TestData/ValidYaml-CI/Screen/with-component-instance.pa.yaml [hunk 1]
```diff
@@ -4,6 +4,7 @@ Hello:
   - This is custom component:
       Control: Component
       ComponentName: ComponentDefinition_1
+      ComponentLibraryUniqueName: MyComponentLibrary
```
Comment (petrochuk) [inline src/Persistence.Tests/_TestData/ValidYaml-CI/Screen/with-component-instance.pa.yaml:7]: Do we have test with and without this?

**Review by petrochuk — APPROVED** [review]

File: src/Persistence.Tests/Yaml/RoundTripTests.cs [hunk 2]
```diff
@@ -92,6 +92,7 @@ public void Screen_With_Component_Instance(string path, bool isControlIdentifier
         customControl.Should().NotBeNull();
         customControl.Name.Should().Be("This is custom component");
         customControl.ComponentName.Should().Be("ComponentDefinition_1");
+        customControl.ComponentLibraryUniqueName.Should().Be("MyComponentLibrary");
```
Comment (petrochuk) [inline src/Persistence.Tests/Yaml/RoundTripTests.cs:95]: Can we add test when this field is optional?

Comment (mizrael) [reply src/Persistence.Tests/Yaml/RoundTripTests.cs:95]: good idea

File: src/Persistence.Tests/_TestData/ValidYaml-CI/Screen/with-component-instance.pa.yaml (hunk 1 above)
Comment (mizrael) [reply src/Persistence.Tests/_TestData/ValidYaml-CI/Screen/with-component-instance.pa.yaml:7]: just added

**Review by shshetty-microsoft — APPROVED** [review]

**Pull request merged**